gluonnlp/data/candidate_sampler.py

```diff
@@ -23,6 +23,31 @@
 import mxnet as mx
 import numpy as np
 
+try:
+    from numba import njit
+    numba_njit = njit(nogil=True)
+except ImportError:
+
+    def numba_njit(func):
+        return func
+
+
+@numba_njit
+def _candidates_mask(negatives, true_samples, true_samples_mask):
```

Comment:
@eric-haibin-lin you mentioned you also need the accidental hits masking feature. Please take a look if this would work for you and  let me  know if you have any suggestions.

Reply:
For my case my true and negatives are all reshaped to 1-D, and I just do a simple broadcast comparison. 
https://github.com/apache/incubator-mxnet/blob/master/example/rnn/large_word_lm/model.py#L115-L118

Reply:
I'll move the candidate sampler for word embedding learning to the scripts folder, similar to your PR.